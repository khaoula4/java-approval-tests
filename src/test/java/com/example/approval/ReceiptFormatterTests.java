package com.example.approval;

import java.util.List;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class ReceiptFormatterTests {

    @Test
    void whenFormatReceipt() {
        final List<ReceiptItem> sampleItems = List.of(
                new ReceiptItem("Tea", 2, 1.50),
                new ReceiptItem("Coffee", 1, 3.20),
                new ReceiptItem("Cookie", 3, 0.80)
        );
        final String orderId = "500";
        final ReceiptFormatter receiptFormatter = new ReceiptFormatter();
        final String generatedReceiptText = receiptFormatter.formatReceipt(sampleItems, orderId);

        Approvals.verify(generatedReceiptText);
    }

    @Test
    void whenFormatReceipt_givenCombinations() {
        final String[] orderIds = { "500", "600" };

        final List<ReceiptItem> itemsA = List.of(
                new ReceiptItem("Tea", 2, 1.50),
                new ReceiptItem("Cookie", 3, 0.80));

        final List<ReceiptItem> itemsB = List.of(
                new ReceiptItem("Coffee", 1, 3.20),
                new ReceiptItem("Muffin", 2, 2.10));

        final List<List<ReceiptItem>> typeSafeList = List.of(itemsA, itemsB);
        @SuppressWarnings("unchecked") final List<ReceiptItem>[] itemSets = typeSafeList.toArray(new List[0]);

        final ReceiptFormatter receiptFormatter = new ReceiptFormatter();

        CombinationApprovals.verifyAllCombinations(
                (String orderId, List<ReceiptItem> items) -> receiptFormatter.formatReceipt(items, orderId), orderIds, itemSets);

    }
}
