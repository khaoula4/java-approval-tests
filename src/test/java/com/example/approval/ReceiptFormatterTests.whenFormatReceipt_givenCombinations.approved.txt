[500, [ReceiptItem[productName=Tea, quantity=2, unitPrice=1.5], ReceiptItem[productName=Cookie, quantity=3, unitPrice=0.8]]] => RECEIPT FOR ORDER 500 :

Tea x2 @ 1,50 = 3,00
Cookie x3 @ 0,80 = 2,40
TOTAL = 5,40
 
[500, [ReceiptItem[productName=Coffee, quantity=1, unitPrice=3.2], ReceiptItem[productName=Muffin, quantity=2, unitPrice=2.1]]] => RECEIPT FOR ORDER 500 :

Coffee x1 @ 3,20 = 3,20
Muffin x2 @ 2,10 = 4,20
TOTAL = 7,40
 
[600, [ReceiptItem[productName=Tea, quantity=2, unitPrice=1.5], ReceiptItem[productName=Cookie, quantity=3, unitPrice=0.8]]] => RECEIPT FOR ORDER 600 :

Tea x2 @ 1,50 = 3,00
Cookie x3 @ 0,80 = 2,40
TOTAL = 5,40
 
[600, [ReceiptItem[productName=Coffee, quantity=1, unitPrice=3.2], ReceiptItem[productName=Muffin, quantity=2, unitPrice=2.1]]] => RECEIPT FOR ORDER 600 :

Coffee x1 @ 3,20 = 3,20
Muffin x2 @ 2,10 = 4,20
TOTAL = 7,40
 
